common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/OneForOneBlockFetcher.java

```diff
@@ -76,12 +83,76 @@ public OneForOneBlockFetcher(
       TransportConf transportConf,
       DownloadFileManager downloadFileManager) {
     this.client = client;
-    this.openMessage = new OpenBlocks(appId, execId, blockIds);
     this.blockIds = blockIds;
     this.listener = listener;
     this.chunkCallback = new ChunkCallback();
     this.transportConf = transportConf;
     this.downloadFileManager = downloadFileManager;
+    if (isShuffleBlocks(blockIds)) {
+      analyzeBlockIds(blockIds);
+      this.message = new FetchShuffleBlocks(
```

Comment:
can we just have a method to create `FetchShuffleBlocks`? We should not add class members unnecessarily.

Reply:
Yep, sorry for less checking here, done in 6d7f1f3.